modules/openapi-generator/src/main/resources/python/python-experimental/model_templates/method_init_composed.mustache

```diff
@@ -34,9 +28,23 @@
         self._composed_instances = composed_info[0]
         self._var_name_to_model_instances = composed_info[1]
         self._additional_properties_model_instances = composed_info[2]
+        unused_args = composed_info[3]
 
 {{#requiredVars}}
         self.{{name}} = {{name}}
 {{/requiredVars}}
         for var_name, var_value in six.iteritems(kwargs):
-            setattr(self, var_name, var_value)
\ No newline at end of file
+            if var_name in unused_args:
+                if self._configuration is not None and self._configuration.discard_unknown_keys:
+                    # discard variable.
+                    pass
+                else:
+                    # This should not happen, any validation problem should have been raised in
+                    # validate_get_composed_info 
+                    raise ApiValueError(
+                        "Invalid input arguments input when making an instance of "
+                        "class %s. Not all inputs were used. The unused input data "
+                        "is %s" % (self.__class__.__name__, unused_args)
+                    )
```

Comment:
Can you keep the exception raising in validate_get_composed_info?
That makes it so we only need to look there to understand what is going wrong vs 2 or more locations.
Also, the model class currently automatically throws an exception when attempting to assign to an unknown variable so we should be okay here.

Reply:
> Can you keep the exception raising in validate_get_composed_info?
I didn't change that. It's still raising the exception. One thing I could do here is do nothing, because validate_get_composed_info has already raised the exception.

> That makes it so we only need to look there to understand what is going wrong vs 2 or more locations.
> Also, the model class setattr currently automatically throws an exception when attempting to assign to an unknown variable so we should be okay here.

Reply:
Can you also add code to cover the non-composed schemas use case?

Reply:
I just refactored a bit.